source/production/F0.Minesweeper.Logic/Minefield.cs

```diff
@@ -1,33 +1,98 @@
+using System;
+using System.Collections.Generic;
 using F0.Minesweeper.Logic.Abstractions;
+using F0.Minesweeper.Logic.Mineplacer;
 
 namespace F0.Minesweeper.Logic
 {
 	internal sealed class Minefield : IMinefield
 	{
 		private readonly uint width, height, mineCount;
+		private IMineplacer mineplacer;
+		private bool isFirstUncover = true;
+		private bool[,] minefield;
+		private string DebugVisualization
+		{
+			get
+			{
+				System.Text.StringBuilder sb = new System.Text.StringBuilder();
+				for (int j = 0; j < height; j++)
```

Comment:
To be in sync with other code, let's have the outer loop iterate _x/width_ and the inner loop _y/height_.

Reply:
I had this like you proposed _at first_, but it sadly doesn't work that way here 😅
As the Stringbuilder builds the string from left to right (_most_ of the time... at least in our culture 😂) , we need to loop over the x/width axis first (inner loop) and then repeat this y/height times (outer loop).
Otherwise, we get a 90° rotated Minefield in the Property.

But awesome and thanks for noticing it 😄

Reply:
oh yes - I totally did not think about that
perhaps an indicator to make that property internal and add a Unit-Test for that